homeassistant/components/foscam/camera.py

```diff
@@ -53,10 +53,13 @@ def __init__(self, device_info):
         self._foscam_session = FoscamCamera(
             ip_address, port, self._username, self._password, verbose=False)
 
-        self._media_port = None
+        self._rtsp_port = None
         result, response = self._foscam_session.get_port_info()
         if result == 0:
-            self._media_port = response['mediaPort']
+            if 'rtspPort' in response.keys():
```

Comment:
This is just preference, but you could do `self._rtsp_port = response.get('rtspPort') or response.get('mediaPort')`

Then if neither exist, it will still default to None, and fall back to MJPEG

Reply:
tested and it works. had to throw a \ in there to keep the line under 80 chars long.